### PR TITLE
adds comment to page

### PR DESCRIPTION
**Pull request by sharifmaryam**
HTML Forms part of tutorial. Added a comments portion and removed the original button from before.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,19 +28,26 @@
 public class DataServlet extends HttpServlet {
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (sylvia-hoang) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: No need to remove the doGet() function! You just have to add doPost() as a new function to the servlet. The SubtractionServlet.java file in the walkthrough contains both, as an example.

All you need to do to your doGet() function is to remove the step where you add your hardcoded messages, which you already did.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -27,19 +27,4 @@ function addRandomGreeting() {
...
-        messagesListElement.appendChild(createListElement(json[i]));
-    }
-    });
-}
 
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
-}
```
Comment (sylvia-hoang) [inline portfolio/src/main/webapp/script.js]: Please keep them!

**Review by sylvia-hoang — CHANGES_REQUESTED** [review]
Remember to compare this branch to master now that you merged the other PR. 

Also, have you try running the dev server and see how this works?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (sharifmaryam) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: So are the doGet() and doPost() functions going to be the same?

Comment (sylvia-hoang) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: Nope, they're different things, as I explained to you in the chat.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -29,9 +32,8 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ArrayList<String> messages = new ArrayList<String>();
```
Comment (sylvia-hoang) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Initialize `messages` outside of both doGet() and doPost() functions. Only do this once.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -29,9 +32,8 @@
...
-    messages.add("Hello");
-    messages.add("Goodbye");
-    messages.add("See you later");
+    String comment = getParameter(request, "textInput", "nothing");
+    messages.add(comment);
```
Comment (sylvia-hoang) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Remove these lines. This is the operation done by doPost(). All doGet() needs to do is the rest of this method that you already have.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -41,5 +43,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<String> messages = new ArrayList<String>();
+    String comment = getParameter(request, "textInput", "nothing");
+    messages.add(comment);
+
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
 
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
```
Comment (sylvia-hoang) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: No need to put the doGet() stuff here. Please remove and readd `response.sendRedirect("/index.html");`. You need to tell the page to redirect there, so the UI can be rerendered and doGet() method gets triggered.

**Review by sylvia-hoang — APPROVED** [review]

**Pull request merged**